localstack/services/sns/sns_listener.py

```diff
@@ -846,18 +890,22 @@ def create_sqs_message_attributes(subscriber, attributes):
 
     message_attributes = {}
     for key, value in attributes.items():
-        attribute = {"DataType": value["Type"]}
-        if value["Type"] == "Binary":
-            attribute["BinaryValue"] = base64.decodebytes(to_bytes(value["Value"]))
-        else:
-            attribute["StringValue"] = str(value.get("Value", ""))
+        if value.get("Type"):  # attributes got from get_message_attributes()
+            attribute = {"DataType": value["Type"]}
+            if value["Type"] == "Binary":
+                attribute["BinaryValue"] = base64.decodebytes(to_bytes(value["Value"]))
+            else:
+                attribute["StringValue"] = str(value.get("Value", ""))
 
-        message_attributes[key] = attribute
+            message_attributes[key] = attribute
+        else:  # attribute get from parse_urlencoded_data()
+            message_attributes[key] = value
 
     return message_attributes
 
 
 def get_message_attributes(req_data):
+    # TODO : Use parse_urlencoded_data() instead of this
```

Comment:
Could this be resolved? Otherwise we end up with two methods being used doing seemingly the same.

Reply:
Hi @baermat , yes it can be resolved and i think it can be completely removed. We might need to refactor the places where `get_message_attributes` is used, since it is not in scope of this PR I left it to do in the next iterations.

Reply:
Sounds good - let's merge the PR now to get the functionality out to the users, and then tackle this in the next iteration.. 👍